Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/NonCompositedFrameRenderer.cpp

```diff
@@ -71,6 +74,28 @@ NonCompositedFrameRenderer::~NonCompositedFrameRenderer()
     m_surface->willDestroyCompositingRunLoop();
 }
 
+void NonCompositedFrameRenderer::addDirtyRect(const IntRect& rect)
+{
+#if ENABLE(DAMAGE_TRACKING)
+    if (m_frameDamage)
+        m_frameDamage->add(rect);
+#else
+    UNUSED_PARAM(rect);
+#endif
+}
+
+#if ENABLE(DAMAGE_TRACKING)
+void NonCompositedFrameRenderer::resetFrameDamage()
+{
+    Ref webPage = m_webPage.get();
+    if (webPage->corePage()->settings().propagateDamagingInformation()) {
+        m_frameDamage = webPage->corePage()->settings().unifyDamagedRegions()
+            ? std::make_optional<Damage>(webPage->bounds(), Damage::Mode::BoundingBox)
+            : std::make_optional<Damage>(webPage->bounds(), Damage::Mode::Rectangles);
```

Comment:
I think this is easier to read as a single make optional and only make conditional the mode parameter